rust/migration/src/m20231213_194935_create_blob_table.rs

```diff
@@ -0,0 +1,141 @@
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(
+                Table::create()
+                    .table(BlobStore::Table)
+                    .col(
+                        ColumnDef::new(BlobStore::Id)
+                            .integer()
+                            .not_null()
+                            .auto_increment()
+                            .primary_key(),
+                    )
+                    .col(ColumnDef::new(BlobStore::Name).string().not_null())
+                    .to_owned(),
+            )
+            .await?;
+
+        // Insert basic backing stores since they must always exist.
+        let insert = Query::insert()
+            .into_table(BlobStore::Table)
+            .columns([BlobStore::Id, BlobStore::Name])
+            .values_panic([1.into(), "local".into()])
+            .values_panic([2.into(), "artifactory".into()])
+            .values_panic([3.into(), "aws".into()])
+            .values_panic([4.into(), "gcs".into()])
+            .values_panic([5.into(), "azure".into()])
+            .to_owned();
+
+        manager.exec_stmt(insert).await?;
+
+        manager
+            .create_table(
+                Table::create()
+                    .table(Blob::Table)
+                    .col(ColumnDef::new(Blob::Id).string().not_null().primary_key())
+                    .col(ColumnDef::new(Blob::Store).integer().not_null())
```

Comment:
I think you probably want a blob url here, not just the store type. Just knowing the store type doesn't help. You need to know the store type and configuration details of the store to know anything useful. I'd argue we should either drop this or make it a full URL